### PR TITLE
Created config scripts loading, defaults to scripts/config.lua

### LINKED ISSUE #13: Window resizing enhancements
Window resizing needs min/max size checking and to make it ratio based.

### PR DESCRIPTION
**Pull request by adam4813**
Tweaked order of render system initialization to support loading config before attempting to create GL objects.

This config script can be acceessed to get values such as graphics options.

Closes #13

### CODE REVIEW

**Review by Meisaka — APPROVED** [review]
This mostly seems okay, the constants for default window width and height are a bit inconsistent and kind of spread out, which could cause weirdness, one option would be moving the defaults to a single header file.
I'm also not sure what the benefit of having aspect be configurable is, I would have probably just computed the aspect from the current window size.

File: client/game.cpp [hunk 1]
```diff
@@ -45,11 +47,17 @@ namespace tec {
 						server_connection.StartSync();
 					});
 			});
+	}
 
-		this->rs.SetViewportSize(viewport_width, viewport_height);
+	void Game::Startup() {
+		this->rs.Startup();
+		const unsigned int window_width = this->config_script->environment.get_or("window_width", 800);
+		const unsigned int window_height = this->config_script->environment.get_or("window_height", 800);
```
[the remark this answers is not in the document]
Comment (adam4813) [reply client/game.cpp:55]: Ah, good call out!

File: client/main.cpp [hunk 2]
```diff
@@ -51,29 +51,42 @@ auto ParseLogLevel(int argc, char* argv[]) {
...
-	if (!os.InitializeWindow(1024, 768, "TEC 0.1", 4, 0)) {
+	if (!os.InitializeWindow(window_width, window_height, "TEC 0.1", 4, 0)) {
 		log->info("Exiting. The context wasn't created properly please update drivers and try again.");
 		exit(1);
 	}
+
+	const std::string aspect_ratio = game.config_script->environment.get_or("aspect_ratio", ASPECT_RATIO);
+	auto numer = stoi(aspect_ratio.substr(0, aspect_ratio.find(':')));
+	auto denom = stoi(aspect_ratio.substr(aspect_ratio.find(':') + 1));
+	os.SetWindowAspectRatio(numer, denom);
```
[the remark this answers is not in the document]
Comment (adam4813) [reply client/main.cpp:80]: Would could derive the default from it yes.

**Comment by adam4813** [comment]
I put it as configurable just because it wasn't hard to do. The user might want to set the AR but not a width/height for example.

**Comment by adam4813** [comment]
@Meisaka Remove the aspect ratio config and made it a calculated value. A future enhancement could be to add a config to disable a fixed aspect ratio.

**Review by Meisaka — APPROVED** [review]
I don't see any glaring issues.
might want to revisit `CalculateAspectRatioString` in the future though

**Comment by adam4813** [comment]
Yep that's why I created a function to isolate it. The interface is good the implementation might need to be improved.

**Pull request merged**